### PR TITLE
Added documentation examples for `use-set-for-membership`.

### PR DESCRIPTION
**Pull request by mbyrnepr2**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Part of https://github.com/PyCQA/pylint/issues/5953

### CODE REVIEW

File: doc/data/messages/u/use-set-for-membership/bad.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+if "banana" in ["apple", "banana", "cantaloupe"]:  # [use-set-for-membership]
+    pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/u/use-set-for-membership/bad.py:2]:
```suggestion
def is_a_fruit_a_cat_could_eat(fruit: str) -> bool:
    return fruit not in ["cherry", "grapes"]  # [use-set-for-membership]
```

File: doc/data/messages/u/use-set-for-membership/good.py [hunk 2]
```diff
@@ -1 +1,2 @@
-# This is a placeholder for correct code for this message.
+if "banana" in {"apple", "banana", "cantaloupe"}:
+    pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/u/use-set-for-membership/good.py:2]:
```suggestion
def is_a_fruit_a_cat_could_eat(fruit: str) -> bool:
    return fruit not in {"cherry", "grapes"}
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
I think some examples make more sense if it's a function and one of the value is unknown (especially code with boolean that would otherwise always have the same value).

**Comment by mbyrnepr2** [comment]
> I think some examples make more sense if it's a function and one of the value is unknown (especially code with boolean that would otherwise always have the same value).

Yes I like it. Thanks @Pierre-Sassoulas.

File: doc/data/messages/u/use-set-for-membership/bad.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+def is_a_fruit_a_cat_could_eat(fruit: str) -> bool:
+    return fruit not in ["cherry", "grapes"]  # [use-set-for-membership]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/u/use-set-for-membership/bad.py:2]:
```suggestion
def fruit_is_dangerous_for_cat(fruit: str) -> bool:
    """This list is only a silly example, don't make decision regarding your cat diet based on it."""
    return fruit in ["cherry", "grapes"]  # [use-set-for-membership]
```

File: doc/data/messages/u/use-set-for-membership/good.py [hunk 4]
```diff
@@ -1 +1,2 @@
-# This is a placeholder for correct code for this message.
+def is_a_fruit_a_cat_could_eat(fruit: str) -> bool:
+    return fruit not in {"cherry", "grapes"}
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/u/use-set-for-membership/good.py:2]:
```suggestion
def fruit_is_dangerous_for_cat(fruit: str) -> bool:
    """This list is only a silly example, don't make decision regarding your cat diet based on it."""
    return fruit in {"cherry", "grapes"}
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
My bad regarding the example: I think if it's ``not in`` there's no benefit to using a set because you will need to check all the iterable anyway and pylint rightly do not complain about it.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**